cmake/pcl_find_cuda.cmake

```diff
@@ -31,25 +22,30 @@ if(CUDA_FOUND)
   # https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#deprecated-features
   
   if(NOT ${CUDA_VERSION_STRING} VERSION_LESS "11.0")
-    set(__cuda_arch_bin "3.5 3.7 5.0 5.2 5.3 6.0 6.1 6.2 7.0 7.2 7.5 8.0")
+    set(__cuda_arch_bin "35;37;50;52;53;60;61;62;70;72;75;80")
   elseif(NOT ${CUDA_VERSION_STRING} VERSION_LESS "10.0")
-    set(__cuda_arch_bin "3.0 3.2 3.5 3.7 5.0 5.2 5.3 6.0 6.1 6.2 7.0 7.2 7.5")
+    set(__cuda_arch_bin "30;32;35;37;50;52;53;60;61;62;70;72;75")
   elseif(NOT ${CUDA_VERSION_STRING} VERSION_LESS "9.0")
-    set(__cuda_arch_bin "3.0 3.2 3.5 3.7 5.0 5.2 5.3 6.0 6.1 6.2 7.0 7.2")
+    set(__cuda_arch_bin "30;32;35;37;50;52;53;60;61;62;70;72")
   endif()
 
-  set(CUDA_ARCH_BIN ${__cuda_arch_bin} CACHE STRING "Specify 'real' GPU architectures to build binaries for, BIN(PTX) format is supported")
-
-  set(CUDA_ARCH_PTX "" CACHE STRING "Specify 'virtual' PTX arch to build PTX intermediate code for. Example: 1.0 1.2 or 10 12")
-  #set(CUDA_ARCH_PTX "1.1 1.2" CACHE STRING "Specify 'virtual' PTX arch to build PTX intermediate code for. Example: 1.0 1.2 or 10 12")
-
-  # Guess this macros will be included in cmake distributive
-  include(${PCL_SOURCE_DIR}/cmake/CudaComputeTargetFlags.cmake)
-  APPEND_TARGET_ARCH_FLAGS()
-
-  # Prevent compilation issues between recent gcc versions and old CUDA versions
-  list(APPEND CUDA_NVCC_FLAGS "-D_FORCE_INLINES")
+  set(CUDA_ARCH_BIN ${__cuda_arch_bin} CACHE STRING "Specify 'real' GPU architectures to build binaries for")
   
-  # Allow calling a constexpr __host__ function from a __device__ function.
-  list(APPEND CUDA_NVCC_FLAGS "--expt-relaxed-constexpr")
+  if(POLICY CMP0104)
+    cmake_policy(SET CMP0104 NEW)
+    foreach(ver ${CUDA_ARCH_BIN})
+      set(CMAKE_CUDA_ARCHITECTURES "${ver}-real;${ver}-virtual;${CMAKE_CUDA_ARCHITECTURES}")
+    endforeach()
+    set(CMAKE_CUDA_ARCHITECTURES "${CMAKE_CUDA_ARCHITECTURES}")
+    message(STATUS "CMAKE_CUDA_ARCHITECTURES: ${CMAKE_CUDA_ARCHITECTURES}")
+  else()
+    # Generate SASS
+    foreach(ver ${CUDA_ARCH_BIN})
+      set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode arch=compute_${ver},code=sm_${ver}")
+    endforeach()
+    # Generate PTX for last architecture
+    list(GET CUDA_ARCH_BIN -1 ver)
+    set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode arch=compute_${ver},code=compute_${ver}")
+    message(STATUS "CUDA GEN_CODE: ${ver}")
```

Comment:
Thoughts about printing `CMAKE_CUDA_FLAGS`?

Reply:
The variable could be modified at a later stage, should I print next to `CMAKE_CXX_FLAGS` even though, not necessarily all targets is compiled with those flags?

Reply:
I have changed it to print flags, which will also print the GEN_CODE versions.